akka-http/src/main/scala/akka/http/scaladsl/server/RejectionHandler.scala

```diff
@@ -156,6 +157,10 @@ object RejectionHandler {
           complete((BadRequest, "Request is missing required HTTP header '" + headerName + '\''))
       }
       .handle {
+        case InvalidOriginHeaderRejection(invalidOrigin) ⇒
+          complete((Forbidden, s"Origin header value '$invalidOrigin' is not in the same origin"))
```

Comment:
The rejection sounds a bit more general than the message that we do here,
let's adjust message to just say "Illegal Origin header value ..."